tests/src/python/test_qgsrasterfilewriter.py

```diff
@@ -223,6 +226,62 @@ def testGeneratePyramidsInternal(self):
     def testGeneratePyramidsErdas(self):
         return self._testGeneratePyramids(QgsRaster.PyramidsErdas)
 
+    def testWriteAsRawInvalidOutputFile(self):
+        tmpName = "/this/is/invalid/file.tif"
+        source = QgsRasterLayer(os.path.join(self.testDataDir, 'raster', 'byte.tif'), 'my', 'gdal')
+        self.assertTrue(source.isValid())
+        provider = source.dataProvider()
+        fw = QgsRasterFileWriter(tmpName)
+
+        pipe = QgsRasterPipe()
+        self.assertTrue(pipe.set(provider.clone()))
+
+        self.assertEqual(fw.writeRaster(pipe,
+                                        provider.xSize(),
+                                        provider.ySize(),
+                                        provider.extent(),
+                                        provider.crs()), QgsRasterFileWriter.DestProviderError)
+        del fw
```

Comment:
Is there a reason for calling `del` explicitly?

Reply:
hum, copy&paste from previously existing tests. I assume that the original intent was to make sure the target file was properly closed before re-reading it, but AFAICS in QgsRasterFileWriter there's no variable member with the output dataset, so probably useless